contracts/mocks/docs/account/MyFactoryAccountECDSA.sol

```diff
@@ -0,0 +1,37 @@
+// contracts/MyFactoryAccountECDSA.sol
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {MyAccountECDSA} from "./MyAccountECDSA.sol";
+
+/**
+ * @dev An abstract factory contract to create ECDSA accounts on demand.
+ */
+contract MyFactoryAccountECDSA {
+    using Clones for address;
+
+    address private immutable _impl = address(new MyAccountECDSA());
+
+    /// @dev Predict the address of the account
+    function predictAddress(bytes32 salt) public view returns (address) {
+        return _impl.predictDeterministicAddress(salt, address(this));
+    }
+
+    /// @dev Create clone accounts on demand
+    function cloneAndInitialize(bytes32 salt, address signer) public returns (address) {
+        return _cloneAndInitialize(salt, signer);
```

Comment:
Should signer be hashed into the hash. Right now this function can be front-run with the same hash and a different signer.